File: src/ctci/FizzBuzz.java
/*
Given a positive integer N, print all the integers from 1 to N. 
But for multiples of 3 print “Fizz” instead of the number and for the multiples of 5 print “Buzz”. Also for number which are multiple of 3 and 5, prints “FizzBuzz”.

Example
N = 5
Return: [1 2 Fizz 4 Buzz]
*/
public class Solution {
    public ArrayList<String> fizzBuzz(int A) {
        ArrayList<String> result = new ArrayList<>();
        
        //Error handling
        if(A <= 0) return result;
        for(int i = 1; i <= A; i++) {
            if(i % 15 == 0) result.add("FizzBuzz");
            else if(i%3 == 0) result.add("Fizz");
            else if(i%5 == 0) result.add("Buzz");
            else result.add((""+i));
        }
        return result;
    }
}

File: src/ctci/FizzBuzz.py
class Solution:
	# @param A : integer
	# @return a list of strings
    def fizzBuzz_v1(self, A):
        
        result = []
        for x in range(1, A+1):
            if x%15 == 0:
                result.append("FizzBuzz")
            elif x%3 == 0:
                result.append("Fizz")
            elif x%5 == 0:
                result.append("Buzz")
            else:
                result.append(str(x))
        return result

    ## Faster solution
    def fizzBuzz(self, A):
        
        result = list(range(1, A+1))
        for i in range(2, A, 3):
            result[i] = "Fizz"
        for i in range(4, A, 5):
            if result[i] == "Fizz": 
                result[i] += "Buzz"
            else:
                result[i] = "Buzz"
        
        return result
